ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -158,7 +158,13 @@ export function AssetPickerModal({
 
   const [searchQuery, setSearchQuery] = useState('');
   const [debouncedSearchQuery, setDebouncedSearchQuery] = useState(searchQuery);
-  const debouncedSetSearchQuery = debounce(setDebouncedSearchQuery, 200);
+  const debouncedSetSearchQuery = useCallback(
+    debounce((value) => {
+      setDebouncedSearchQuery(value);
+    }, 200),
+    [],
+  );
+
   useEffect(() => {
     debouncedSetSearchQuery(searchQuery);
```

Comment:
untested, but for clean up, it'd be something like
```suggestion
    debouncedSetSearchQuery(searchQuery);
    
    // clean up the debounced function on unmount or when `searchQuery` changes
    return () => {
      debouncedSetSearchQuery.cancel();
    };
```

Reply:
Updated the PR with this and also cancellation for the abortController

Reply:
Wouldn't this cause `debouncedSetSearchQuery.cancel()` to run before every `debouncedSetSearchQuery` call (line 180) that is triggered by a `searchQuery` update? I think that might interfere with the debounce operation.

Reply:
Ah I think I misunderstood the intention. If we want to prevent `.cancel()` from being called unless `searchQuery` has changed the suggestion above would work.

Reply:
Thanks @davidmurdoch , I will sync with @MajorLift 

> If we want to prevent .cancel() from being called unless searchQuery has changed the suggestion above would work.

This is what we want so I think the suggestion works. Implemented this in the latest commit